### PR TITLE
Add masterkey parameters

### PR DESCRIPTION
**Pull request by mtrezza**
Adds a property to Parse Config parameter to make it only retrievable with master key.

### CODE REVIEW

**Comment by mtrezza** [comment]
This feature requires https://github.com/parse-community/parse-server/pull/5954.

**Comment by davimacedo** [comment]
@mtrezza nice job! Could you please just centralize the field and set the background color to the same of the other fields? Below how it current is:
Screen Shot 2019-08-20 at 11 12 41 PM

It would be good if it had the same style of the other similar fields. An example below:
Screen Shot 2019-08-20 at 11 14 24 PM

**Comment by davimacedo** [comment]
Also.. in the other places this field is used the labels are `No` and `Yes`

**Comment by mtrezza** [comment]
@davimacedo Done, looks like this now:

Screen Shot 2019-08-21 at 12 15 47

File: src/dashboard/Data/Config/ConfigDialog.react.js [hunk 1]
```diff
@@ -228,6 +232,21 @@ export default class ConfigDialog extends React.Component {
               description='Use this to configure your app. You can change it at any time.' />
           }
           input={EDITORS[this.state.type](this.state.value, (value) => { this.setState({ value }) })} />
+        <Field
+          label={
+            <Label
+              text='Requires master key?'
+              description='The parameter is returned only when requested with the master key. You can change it at any time.' />
```
Comment (TomWFox) [inline src/dashboard/Data/Config/ConfigDialog.react.js]: Probably good to be super explicit...
```suggestion
              description='When true the parameter is returned only when requested with the master key. You can change it at any time.' />
```

**Comment by mtrezza** [comment]
@TomWFox, as @davimacedo suggested, we are going away from the notion of “true/false” in favor of “yes/no”. Do you want to change the wording in your suggestion?

File: src/dashboard/Data/Config/ConfigDialog.react.js (hunk 1 above)
Comment (mtrezza) [reply src/dashboard/Data/Config/ConfigDialog.react.js]: As @davimacedo suggested, we are going away from the notion of “true/false” in favor of “yes/no”. Do you want to change the wording in your suggestion?

Comment (mtrezza) [reply src/dashboard/Data/Config/ConfigDialog.react.js]: As @davimacedo suggested, we are going away from the notion of “true/false” in favor of “yes/no”. Do you want to change the wording in your suggestion?

Comment (mtrezza) [reply src/dashboard/Data/Config/ConfigDialog.react.js]: As @davimacedo suggested, we are going away from the notion of “true/false” in favor of “yes/no”. Do you want to change the wording in your suggestion?

Comment (TomWFox) [reply src/dashboard/Data/Config/ConfigDialog.react.js]: Sure

**Comment by TomWFox** [comment]
Could you check that the text is displayed correctly with the extra words?

**Comment by mtrezza** [comment]
Actually I am thinking about adding a parse server version check for that new property. So the dashboard is backwards compatible with older parse server versions. @davimacedo which parse server version should I add as the minimum? 3.7.3?

**Comment by TomWFox** [comment]
Lots of changes for the next release so it should be 3.8.0

**Comment by mtrezza** [comment]
> Could you check that the text is displayed correctly with the extra words?

Yes, text fits.

**Comment by mtrezza** [comment]
All changes done, ready for merge I believe.

File: src/dashboard/Data/Config/ConfigDialog.react.js [hunk 2]
```diff
@@ -228,6 +233,30 @@ export default class ConfigDialog extends React.Component {
               description='Use this to configure your app. You can change it at any time.' />
           }
           input={EDITORS[this.state.type](this.state.value, (value) => { this.setState({ value }) })} />
+        
+        {
+          /*
+            Add `Requires master key` field if parse-server version >= 3.8.0,
+            that is the minimum version that supports this feature.
+          */
+          semver.valid(this.props.parseServerVersion) && semver.gte(this.props.parseServerVersion, '3.8.0')
```
Comment (davimacedo) [inline src/dashboard/Data/Config/ConfigDialog.react.js]: We use to check the [server info](https://github.com/parse-community/parse-server/blob/master/src/Routers/FeaturesRouter.js) to decide if a feature is available or not at Parse Dashboard. What do you think?

Comment (mtrezza) [reply src/dashboard/Data/Config/ConfigDialog.react.js]: I took this approach from here:
https://github.com/parse-community/parse-dashboard/blob/169e13b2328fc5adff2afffa2f6e855eb7cc638e/src/dashboard/Data/Browser/AddColumnDialog.react.js#L217-L218

Comment (mtrezza) [reply src/dashboard/Data/Config/ConfigDialog.react.js]: And as I understand it, `app.serverInfo` is set by calling the `/serverInfo` endpoint when the dashboard is launched:
https://github.com/parse-community/parse-dashboard/blob/169e13b2328fc5adff2afffa2f6e855eb7cc638e/src/dashboard/Dashboard.js#L129-L131

**Comment by davimacedo** [comment]
@dplewis do you want to take a look as well?

**Comment by mtrezza** [comment]
@davimacedo This feature has been released with parse server 3.8. Can you create a dashboard release so the feature can be used, since there are also no open dashboard PRs?

**Comment by davimacedo** [comment]
I've just sent the PR for the new version: https://github.com/parse-community/parse-dashboard/pull/1263/files

**Pull request merged**